### ISSUE TITLE
Is it possible to embed one protocol into another?

### ISSUE THREAD

**Issue by itavero** [robotframework/Rammbock#34]
So we have a TCP-based protocol (lets call it protocol A for now), that basically adds some additional fields to address specific components within our product.
These components may speak different protocols, so the payload of that first protocol can be a number of different protocols (protocol B, C, D, ...).

Is it possible to tell Rammbock that protocol B is prefixed with the header of protocol A?

**Comment by jussimalinen** [comment]
Not really... But maybe something like that can be emulated by different messagetypes. So the actual protocol is the A, but then different messagetypes are the subprotocols.

Difficult to comment more than that without more details.

**Comment by itavero** [comment]
The "subprotocols" also have different message types, so that would get messy pretty quick.
We are using Suitcase for now to generate and dissect our protocol messages.
